loopy/kernel/tools.py

```diff
@@ -313,7 +317,9 @@ def op(self, set, op_name, op, args):
         bucket = self.cache.setdefault(hashval, [])
```

Comment:
This is strange to me:

- `op_name` seems redundant
- We could just make `op` part of the key. Maybe `args`, too, if we demand hashability.

Reply:
Thanks for taking a look, applied the suggestion.